fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java

```diff
@@ -0,0 +1,243 @@
+import com.google.common.collect.Lists;
+import com.starrocks.analysis.BrokerDesc;
+import com.starrocks.analysis.DescriptorTable;
+import com.starrocks.common.DdlException;
+import com.starrocks.common.UserException;
+import com.starrocks.fs.HdfsUtil;
+import com.starrocks.proto.PGetFileSchemaResult;
+import com.starrocks.proto.PScalarType;
+import com.starrocks.proto.PSlotDescriptor;
+import com.starrocks.proto.PTypeNode;
+import com.starrocks.rpc.BackendServiceClient;
+import com.starrocks.rpc.PGetFileSchemaRequest;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.sql.ast.ImportColumnDesc;
+import com.starrocks.system.Backend;
+import com.starrocks.thrift.TBrokerFileStatus;
+import com.starrocks.thrift.TBrokerRangeDesc;
+import com.starrocks.thrift.TBrokerScanRange;
+import com.starrocks.thrift.TBrokerScanRangeParams;
+import com.starrocks.thrift.TColumn;
+import com.starrocks.thrift.TFileFormatType;
+import com.starrocks.thrift.TFileType;
+import com.starrocks.thrift.TGetFileSchemaRequest;
+import com.starrocks.thrift.THdfsProperties;
+import com.starrocks.thrift.TNetworkAddress;
+import com.starrocks.thrift.TPrimitiveType;
+import com.starrocks.thrift.TScanRange;
+import com.starrocks.thrift.TTableDescriptor;
+import com.starrocks.thrift.TTableFunctionTable;
+import com.starrocks.thrift.TTableType;
+import org.apache.thrift.TException;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Future;
+
+public class TableFunctionTable extends Table {
+
+    private static final String PROPERTY_PATH = "path";
+    private static final String PROPERTY_FORMAT = "format";
+
+    private String path;
+    private String format;
+    private Map<String, String> properties;
+
+    private List<TBrokerFileStatus> fileStatuses = Lists.newArrayList();
+
+    public TableFunctionTable(Map<String, String> properties) throws DdlException {
+        super(TableType.TABLE_FUNCTION);
+        super.setId(-1);
+        super.setName("table_function_table");
+        this.properties = properties;
+        parseProperties();
+
+        parseFiles();
+
+        super.setNewFullSchema(getFileSchema());
+    }
+
+    public List<TBrokerFileStatus> fileList() {
+        return fileStatuses;
+    }
+
+    @Override
+    public Map<String, String> getProperties() {
+        return properties;
+    }
+
+    @Override
+    public TTableDescriptor toThrift(List<DescriptorTable.ReferencedPartitionInfo> partitions) {
+        TTableFunctionTable tTbl = new TTableFunctionTable();
+        tTbl.setLocation(path);
+
+        List<TColumn> tColumns = Lists.newArrayList();
+
+        for (Column column : getBaseSchema()) {
+            tColumns.add(column.toThrift());
+        }
+        tTbl.setColumns(tColumns);
+
+        TTableDescriptor tTableDescriptor = new TTableDescriptor(id, TTableType.FILE_TABLE, fullSchema.size(),
+                0, "", "");
```

Comment:
setting some pseudo table/db name may help us to diagnose problem

Reply:
Updated. `_table_function_table` and `_table_function_db` are set here.